calc/event_sequencer.go

```diff
@@ -700,6 +716,86 @@ func (buf *EventSequencer) flushNamespaces() {
 	log.Debug("Done flushing Namespaces")
 }
 
+func (buf *EventSequencer) OnVTEPUpdate(update *proto.VXLANTunnelEndpointUpdate) {
+	// We trust the caller not to send us an update with nil ID, so safe to dereference.
+	node := update.Node
+	log.WithFields(log.Fields{
+		"id": node,
+	}).Debug("VTEP update")
+	buf.pendingVTEPDeletes.Discard(node)
+	buf.pendingVTEPUpdates[node] = update
+}
+
+func (buf *EventSequencer) OnVTEPRemove(dst string) {
+	log.WithFields(log.Fields{
+		"dst": dst,
+	}).Debug("VTEP removed")
+	delete(buf.pendingVTEPUpdates, dst)
+	if buf.sentVTEPs.Contains(dst) {
+		buf.pendingVTEPDeletes.Add(dst)
+	}
+}
+
+func (buf *EventSequencer) flushVTEPs() {
```

Comment:
I think we'll want to reorg this bit so that we can guarantee ordering here:
- Send VTEP adds
- Send route deletes
- Send route adds
- Send VTEP deletes